package/MDAnalysis/lib/picklable_file_io.py

```diff
@@ -0,0 +1,188 @@
+
+    This class provides a file-like object (as returned by :func:`open`, namely :class:`io.FileIO`)
+    that, unlike standard Python file objects, can be pickled. Only read mode is supported.
+
+    When the file is pickled, filename and position of the open file handle in the file
+    are saved. On unpickling, the file is opened by filename and the file is
+    seeked to the saved position. This means that for a successful unpickle, the original
+    file still has to be accessible with its filename.
+
+    Example
+    -------
+    ::
+        file = FileIOPicklable('filename')
+        file.readline()
+        file_pickled = pickle.loads(pickle.dumps(file)
+        assert_equal(file.tell(), file_pickled.tell())
+
+    See Also
+    ---------
+    TextIOPicklable
+    BufferIOPicklable
+    """
+    def __init__(self, name):
+        super().__init__(name, mode='r')
+
+    def __getstate__(self):
+        return self.name, self.tell()
+
+    def __setstate__(self, args):
+        name = args[0]
+        super().__init__(name, mode='r')
+        self.seek(args[1])
+
+
+class BufferIOPicklable(io.BufferedReader):
+    """A picklable buffer object for read-only FilIO object.
+
+    This class provides a buffered :class:`io.BufferedReader` object that can be pickled. Note that this only works in reda mode.
+
+    Example
+    -------
+    ::
+        file = FileIOPicklable('filename')
+        buffer_wrapped = BufferIOPicklable(file)
+
+    See Also
+    ---------
+    FileIOPicklable
+    TextIOPicklable
+    """
+    def __init__(self, raw):
+        super().__init__(raw)
+        self.raw_class = raw.__class__
+
+    def __getstate__(self):
+        return self.raw_class, self.name, self.tell()
+
+    def __setstate__(self, args):
+        raw_class = args[0]
+        name = args[1]
+        raw = raw_class(name)
+        super().__init__(raw)
+        self.seek(args[2])
+
+
+class TextIOPicklable(io.TextIOWrapper):
+    """Character and line based picklable file-like object.
+
+    This class provides a file-like :class:`io.TextIOWrapper` object that can
+    be pickled. Note that this only works in read mode.
+
+    Example
+    -------
+    ::
+        file = FileIOPicklable('filename')
+        text_wrapped = TextIOPicklable(file)
+
+    See Also
+    ---------
+    FileIOPicklable
+    BufferIOPicklable
+    """
+    def __init__(self, raw):
+        super().__init__(raw)
+        self.raw_class = raw.__class__
+
+    def __getstate__(self):
+        return self.raw_class, self.name, self.tell()
+
+    def __setstate__(self, args):
+        raw_class = args[0]
+        name = args[1]
+        # raw_class is used for further expansion this functionality to
+        # GZip files, which also requires a text wrapper.
+        raw = raw_class(name)
+        super().__init__(raw)
+        self.seek(args[2])
+
+
+def pickle_open(name, mode='rt'):
+    """Open file and return a stream with pickle function implemented.
+
```

Comment:
Before talking about the motivation, say what it _does_. Mention that it returns FileIOPicklable, BufferIOPicklabe, or TextIOPicklabe and that it can also be used as a context manager. It's used instead of :func:`open`.

(Btw, I didn't know that you could just use the function as a context manager... neat – if this also works with `anyopen()` then we could get rid of [`openany()`](https://www.mdanalysis.org/docs/documentation_pages/lib/util.html?highlight=openany#MDAnalysis.lib.util.openany). )